quiche/src/lib.rs

```diff
@@ -3401,7 +3421,7 @@ impl Connection {
     }
 
     fn send_single(
-        &mut self, out: &mut [u8], send_pid: usize, has_initial: bool,
+        &mut self, paths: &mut path::PathMap, out: &mut [u8], send_pid: usize, has_initial: bool,
```

Comment:
What is needed from the connection to send a single packet on a path, could we cache these attributes on the path and just move send_single to the path?